packages/hyperion-autologging/src/ALTriggerFlowlet.ts

```diff
@@ -52,7 +52,7 @@ export function init(options: InitOptions) {
 
   const { channel, flowletManager } = options;
 
-  let currTriggerFlowlet = new flowletManager.flowletCtor('pageload');
+  let currTriggerFlowlet = new flowletManager.flowletCtor('pageload', flowletManager.root);
```

Comment:
`currTriggerFlowlet` did have a root flowlet, just not the "right" root flowlet that the "unexpected root flowlet" check was looking for, right?

Reply:
`currTriggerFlowlet` was the root of many other triggers (e.g. actions). But it itself was not forking the root flowlet. 
The assumption in the code is that for known cases, we fork the root so that we can assign values just to the root and everyone automatically inherits it. 
We already do this for UI events, but had missed doing it here as well.

Reply:
Then should we also update the `FlowletManager` constructor to default to `flowletManager.root` if `parent` is not defined in the constructor arguments? I'm concerned that we may have other gaps or will miss this again when creating new flowlets